lib/vector-core/buffers/src/lib.rs

```diff
@@ -63,20 +74,38 @@ where
         } => {
             let buffer_dir = format!("{}_buffer", id);
 
-            let (tx, rx, acker) =
-                disk::open(&data_dir, &buffer_dir, max_size).map_err(|error| error.to_string())?;
-
+            let (tx, rx, acker) = disk::open(&data_dir, &buffer_dir, max_size, span)
+                .map_err(|error| error.to_string())?;
             let tx = BufferInputCloner::Disk(tx, when_full);
             Ok((tx, rx, acker))
         }
         Variant::Memory {
             max_events,
             when_full,
+            instrument,
+            ..
         } => {
             let (tx, rx) = mpsc::channel(max_events);
-            let tx = BufferInputCloner::Memory(tx, when_full);
-            let rx = Box::new(rx);
-            Ok((tx, rx, Acker::Null))
+            if instrument {
```

Comment:
Is the instrumentation toggle here for our testing in this PR or so users can flag telemetry off for themselves?

Reply:
This is to toggle telemetry for memory buffers and specifically for the memory buffers attached to `transforms`. There is some additional [context here](https://github.com/vectordotdev/vector/pull/9327#discussion_r717887739), but this is mostly to hide implementation details. Currently, users would not be able to flag off sink buffer telemetry.